Support deduplication when using ClusterctlConfig

furkatgofurov7 opened the pull request:
**What this PR does / why we need it**:

<!-- Enter a description of the change and why this change is needed -->

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:


**Special notes for your reviewer**:
If new provider is added through clusterctlconfig:

1. which does not exist in the current config map it will be added as new entry to configmap
2. with new URL and  if provider exists, it only updates the URL of that entry
3. without any changes to URL and if provider exists, it stays as is without any changes.

I am planning to add tests as a follow up PR, since config.go anyways did not have a dedicated test file up until now.


**Checklist**:
<!-- Put an "X" character inside the brackets of each completed task. Some may be optional depending on the PR in which case these can be deleted -->

- [x] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

Danil-Grigorev reviewed (approved):
Can you please add unit tests for the PR? Also you mentioned some kind of issue with bootstrap or control plane provider? Is this resolved? I see no changes since last commit @furkatgofurov7 

LGTM otherwise

alexander-demicev reviewed (approved):
I removed "fixes" from the PR description so issue doesn't close. Tests can be added in a follow-up PR.

furkatgofurov7 commented:
> Can you please add unit tests for the PR? Also you mentioned some kind of issue with bootstrap or control plane provider? Is this resolved? I see no changes since last commit @furkatgofurov7
> 
> LGTM otherwise

No issues with bootstrap after I tested it properly and it is resolved. I will open a follow-up with unit tests addition.
